services/sn_analysis_service/app.py

```diff
@@ -112,55 +138,81 @@ async def run_sn_model(dd):
             model.param_names,
             bounds=bounds,
         )
-        log(f"Number of χ² function calls: {result.ncall}")
-        log(f"Number of degrees of freedom in fit: {result.ndof}")
-        log(f"χ² value at minimum: {result.chisq}")
-        log(f"model parameters: {result.param_names}")
-        log(f"best-fit values: {result.parameters}")
-        log(f"The result contains the following attributes:\n {result.keys()}")
-        log(f'{result["success"]=}')
-
-        if result["success"]:
-
-            with tempfile.NamedTemporaryFile(
-                suffix=".png", prefix="snplot_", dir="/tmp/"
-            ) as f:
-                _ = sncosmo.plot_lc(
-                    data,
-                    model=fitted_model,
-                    errors=result.errors,
-                    model_label=source,
-                    zp=23.9,
-                    figtext=data_dict["resource_id"],
-                    fname=f.name,
-                )
-                f.seek(0)
-                plot_data = io.BytesIO(f.read())
-                f.close()
-                os.remove(f.name)
-                log('made lightcurve plot')
+
+        if result.success:
+            f = tempfile.NamedTemporaryFile(
+                suffix=".png", prefix="snplot_", dir="/tmp/", delete=False
+            )
+            f.close()
+            _ = sncosmo.plot_lc(
+                data,
+                model=fitted_model,
+                errors=result.errors,
+                model_label=source,
+                zp=23.9,
+                figtext=data_dict["resource_id"],
+                fname=f.name,
+            )
+            plot_data = base64.b64encode(open(f.name, "rb").read())
+            local_temp_files.append(f.name)
 
             # make some draws from the posterior (simulating what we'd expect
             # with an MCMC analysis)
             post = rng.multivariate_normal(result.parameters, result.covariance, 10000)
             post = post[np.newaxis, :]
             # create an inference dataset
-            dataset = az.convert_to_inference_data(
+            inference = az.convert_to_inference_data(
                 {x: post[:, :, i] for i, x in enumerate(result.param_names)}
             )
-            results.update({"data": {"posterior": dataset, "plot": plot_data}})
+            f = tempfile.NamedTemporaryFile(
+                suffix=".nc", prefix="inferencedata_", dir="/tmp/"
+            )
+            f.close()
+            inference.to_netcdf(f.name)
+            inference_data = base64.b64encode(open(f.name, 'rb').read())
```

Comment:
Why not BytesIO? Are these files very large?

Reply:
I tried that first. The problem is that we have to send this data in a JSON packet and I was getting unicode errors with BytesIO as the JSON was trying to get serialized by `requests`. This approach of using b64encoding is what we do for thumbnails and seems to work well.

Reply:
The b64encoding is correct, but the temporary file is not necessary if `to_netcdf` is happy dumping to `BytesIO`. You'd keep the reading from the buffer the same.

Reply:
It is not happy dumping to a file-like object. It must be a file.

Reply:
cf.  https://github.com/arviz-devs/arviz/blob/44b908f20372b28b82fffb537c5f69628a99b02e/arviz/data/io_netcdf.py#L46